providers/network/resources/tlsshake/tlsshake.go

```diff
@@ -309,12 +309,24 @@ func (s *Tester) parseAlert(data []byte, conf *ScanConfig) error {
 	return nil
 }
 
+var helloRetryRequestRandom = []byte{
+	0xcf, 0x21, 0xad, 0x74, 0xe5, 0x9a, 0x61, 0x11,
+	0xbe, 0x1d, 0x8c, 0x02, 0x1e, 0x65, 0xb8, 0x91,
+	0xc2, 0xa2, 0x11, 0x16, 0x7a, 0xbb, 0x8c, 0x5e,
+	0x07, 0x9e, 0x09, 0xe2, 0xc8, 0xa8, 0x33, 0x9c,
+}
+
 func (s *Tester) parseServerHello(data []byte, version string, conf *ScanConfig) error {
 	idx := 0
 
-	idx += 2 + 32
-	// handshake tls version (2), which we don't need yet (we will look at it in the extension if necessary)
-	// random (32), which we don't need
+	idx += 2
+	random := data[idx : idx+32]
+	if bytes.Equal(random, helloRetryRequestRandom) {
+		// The server wants us to retry. However, we hardcode a bunch of inputs
+		// so we can't really retry.
+		return errors.New("unexpected HelloRetryRequest")
```

Comment:
Is it possible to extract the key_share or supported_versions from the response extensions?

That might help the next we hit this, and we need to figure out what else besides `secp256r1` is needed.

Reply:
I've added the key share group to the error message. We already seem to do something with supported versions